src/vl-map-layer.js

```diff
@@ -117,11 +122,21 @@ export class VlMapLayer extends VlElement(HTMLElement) {
     }
 
     get _map() {
-        if (this.parentNode) {
-            return this.parentNode.map;
+        if (this._mapElement) {
+            return this._mapElement.map;
         }
     }
 
+    get _mapReady() {
+        if (this._mapElement) {
+            return this._mapElement.ready;
+        }
```

Comment:
als ge await this._mapReady gaat doen op het moment dat this._mapElement er nog niet is, hoe gaat dat dan reageren?

Reply:
of moet ge dan niet beter bij get _mapElement nen error throwen als er geen parentNode blijkt te zijn?

Reply:
Eigenlijk gaat ge altijd een _mapElement hebben, aangezien uw _mapElement gewoon de parent is momenteel. In dat opzicht is de if-test een beetje onnozel en kan die zelfs weg.

Reply:
Ik doe de if-test gewoon weg, aangezien ge altijd de map als parentNode gaat hebben bij correct gebruik.